gossip/src/duplicate_shred_listener.rs

```diff
@@ -0,0 +1,129 @@
+use {
+    crate::{
+        cluster_info::{ClusterInfo, GOSSIP_SLEEP_MILLIS},
+        crds::Cursor,
+        duplicate_shred::DuplicateShred,
+    },
+    std::{
+        sync::{
+            atomic::{AtomicBool, Ordering},
+            Arc,
+        },
+        thread::{self, sleep, Builder, JoinHandle},
+        time::Duration,
+    },
+};
+
+pub trait DuplicateShredHandlerTrait: Send {
```

Comment:
why the need for the trait? we are no longer listening to all the entries right? so there should only b e one handler for DuplicateShred anyway?

Reply:
It's easier to inject a mock handler in the tests this way, so you know when send is called.